src/main/scala/inox/Main.scala

```diff
@@ -215,20 +246,25 @@ trait MainHelpers {
 }
 
 object Main extends MainHelpers {
+  import inox.trees._
 
   def main(args: Array[String]): Unit = {
     val ctx = setup(args)
+    var error: Boolean = false
 
-    val files = ctx.options.findOptionOrDefault(optFiles)
-    if (files.isEmpty) {
-      ctx.reporter.error(s"Input file was not specified.\nTry the --help option for more information.")
-      exit(error = true)
-    } else {
-      var error: Boolean = false
-      for (file <- files; (program, expr) <- tip.Parser(file).parseScript) {
-        import ctx._
-        import program._
+    for {
+      file <- getFilesOrExit(ctx)
+      (program, exprOpt) <- parse(ctx, file)
+    } {
+      import ctx._
+      import program._
 
+      if (ctx.options.findOptionOrDefault(optPrintProgram)) {
+        ctx.reporter.info(s"Program in $file:\n\n")
+        ctx.reporter.info(program)
```

Comment:
I forgot to mention in my previous review, you might want to use `program.asString(PrinterOptions.fromContext(ctx))` here so that printer options are correctly taken into account.

Reply:
Done 👍